packages/hub/services/statuspage-api.ts

```diff
@@ -0,0 +1,124 @@
+import config from 'config';
+import fetch from 'node-fetch';
+
+interface StatuspageComponent {
+  id: string;
+  name: string;
+}
+
+interface StatuspageIncident {
+  id: string;
+  name: string;
+  component_ids: [];
+}
+
+let baseUrl = 'https://api.statuspage.io/v1';
+let pageId = config.get('statuspage.pageId');
+let apiKey = config.get('statuspage.apiKey');
+
+export default class StatuspageApi {
+  async createIncident(componentName: string, name: string) {
+    let componentId = await this.getComponentId(componentName);
+    let unresolvedIncidents = await this.getUnresolvedIncidents();
+    let relatedIncident = this.findIncidentByComponentId(unresolvedIncidents, componentId);
+
+    if (relatedIncident) {
+      return; // An incident for this component is already open
+    }
+
+    return await this.request(`/pages/${pageId}/incidents`, 'POST', {
+      incident: {
+        name,
+        impact: 'minor',
+        status: 'investigating',
+        components: { [componentId]: 'partial_outage' }, // Will update individual components' status
+        component_ids: [componentId], // Will associate incident with the component
+      },
+    });
+  }
+
+  async resolveIncident(componentName: string) {
+    let componentId = await this.getComponentId(componentName);
+    let unresolvedIncidents = await this.getUnresolvedIncidents();
+    let relatedIncident = this.findIncidentByComponentId(unresolvedIncidents, componentId);
```

Comment:
How should this interact with manually created incidents for the same component? It seems like right now there is a possibility that the manually created incidents could get resolved

Reply:
The case I'm wondering about is: 
1. There is a manually created incident for the "Subgraph" component, and also an automatically created one
2. The subgraph is no longer slow so hub tries to resolve the incident. The manually created incident is the one that gets resolved, while the automatically created one remains.

Reply:
i think the compromise here is that if we want to have automatically resolved incidents we need to accept the fact that we can't manually create incidents in the same component.

Reply:
But it also make me wonder what kind of incident we would create manually that wouldn't be detected by the block height delta do you forsee that we need to create

Reply:
That is an interesting question... I guess something like https://thegraph.com/docs/en/developer/create-subgraph-hosted/#non-fatal-errors

Reply:
hmm, we are not actually supporting non-fatal errors though. right now any kind of error will halt the indexing and trigger the block delta check.

Reply:
it seems like a large penalty to pay--but continuing to index with errors results in bad data, and that bad data can feed into other index data which will slowly taint all the data after the bad data was introduced. so failing fast and loud is an easier way for us to find and fix the problems

Reply:
I just noticed there's a `metadata` field we can use to tag automatically created ones, and then only allow automatically resolving these. However this sounds like an edge case we can live with for now and see how this automatisation plays out.